src/Symfony/Component/Security/Core/Security.php

```diff
@@ -57,4 +66,39 @@ public function getToken(): ?TokenInterface
     {
         return $this->container->get('security.token_storage')->getToken();
     }
+
+    public function login(UserInterface $user, AuthenticatorInterface $authenticator = null, string $firewallName = null): void
+    {
```

Comment:
I think it would be nice if the authenticator could be a string, like `form_login`.
Maybe we should add multiple methods like `login(UserInterface $user, ?string $authenticatorName = null, ?string $firewallName = null)` and loginWithAuthenticator(UserInterface $user, AuthenticatorInterface $authenticator)`. What do you think?

Reply:
Sounds good to me. I'd rather keep only `login(UserInterface $user, ?string $authenticatorName = null, ?string $firewallName = null)`, passing the authenticator name/id is enough for the case at hand and avoids the need for retrieving it first

Reply:
Done